runtime/src/main/java/com/flipkart/flux/dao/iface/StatesDAO.java

```diff
@@ -67,4 +67,11 @@
      */
     List findStatesByStatus(ShardId shardId, String stateMachineName, Timestamp fromTime, Timestamp toTime, String taskName, List<Status> statuses);
 
+    /**
+     * Retrieves all states for a particular state-machine-id and like input dependent-event-name.
+     * This query results are specifically used in business logic to make a decision whether to edit EventData or not
```

Comment:
Use case of a method need not be mentioned, first line of the paragraph is enough to understand what this method intend to do.

Reply:
Done.